libs/core/algorithms/include/hpx/parallel/container_algorithms/contains.hpp

```diff
@@ -148,8 +137,7 @@ namespace hpx::parallel { namespace detail {
                 HPX_FORWARD(Proj2, proj2));
 
             return util::detail::algorithm_result<ExPolicy, bool>::get(
-                contains_subrange_helper<ExPolicy, FwdIter1>().get(itr) !=
-                last1);
+                contains_subrange_helper<ExPolicy, FwdIter1>().get(itr, last1));
```

Comment:
I think returning `contains_subrange_helper<ExPolicy, FwdIter1>().get(itr, last1)` would suffice here. 
The return types of your helper function already match with `algorithm_result_t`. Calling that additional `::get` will call `.get()` on your future before returning it, which undermines your latest change on your helper.